build_all/mac/release.sh

```diff
@@ -45,96 +51,138 @@ process_args()
+        python_lib=$python_prefix/lib/libpython${PYTHON_VERSION}m.dylib
+    fi
+
+
+    rm -r -f $build_folder
+    mkdir -p $build_folder
+    pushd $build_folder
+    cmake -Drun_valgrind:BOOL=OFF -DcompileOption_CXX:STRING="-Wno-unused-value" -DcompileOption_C:STRING="-Wno-unused-value" -Drun_e2e_tests:BOOL=OFF -Drun_longhaul_tests=OFF -Duse_amqp:BOOL=ON -Duse_http:BOOL=ON -Duse_mqtt:BOOL=ON -Ddont_use_uploadtoblob:BOOL=OFF -Duse_wsio:BOOL=ON -Drun_unittests:BOOL=OFF -Dbuild_python:STRING=$PYTHON_VERSION -Dbuild_javawrapper:BOOL=OFF -Dno_logging:BOOL=OFF $c_build_root -Dwip_use_c2d_amqp_methods:BOOL=OFF -DPYTHON_LIBRARY=$python_lib -DPYTHON_INCLUDE_DIR=$python_include -Duse_prov_client=OFF -Duse_edge_modules=ON -Dskip_samples=ON
+
+    # Set the default cores
+    CORES=$(sysctl -n hw.ncpu)
+
+    make --jobs=$CORES
+    ctest -j $CORES -C "Debug" --output-on-failure
+
+    popd
+}
 
-make --jobs=$CORES
-ctest -j $CORES -C "Debug" --output-on-failure
+copy_binaries()
+{
+    [ $? -eq 0 ] || exit $?
+    cd $build_root
+
+    echo copy iothub_client library to samples folder
+    cp $build_folder/python/src/iothub_client.dylib ./device/samples/iothub_client.so
+    echo copy iothub_client_mock library to tests folder
+    cp $build_folder/python/test/iothub_client_mock.dylib ./device/tests/iothub_client_mock.so
+    cp $build_folder/python/src/iothub_client.dylib ./device/tests/iothub_client.so
+    cp $build_folder/python_service_client/src/iothub_service_client.dylib ./device/tests/iothub_service_client.so
+
+    echo copy iothub_service_client library to samples folder
+    cp $build_folder/python_service_client/src/iothub_service_client.dylib ./service/samples/iothub_service_client.so
+    echo copy iothub_service_client_mock library to tests folder
+    cp $build_folder/python_service_client/tests/iothub_service_client_mock.dylib ./service/tests/iothub_service_client_mock.so
+    cp $build_folder/python_service_client/src/iothub_service_client.dylib ./service/tests/iothub_service_client.so
+    cp $build_folder/python/src/iothub_client.dylib ./service/tests/iothub_client.so
+}
 
-popd
-########################################################
+run_unit_tests()
+{
+    cd $build_root/device/tests/
+    echo "python${PYTHON_VERSION}" iothub_client_ut.py
+    "python${PYTHON_VERSION}" iothub_client_ut.py
+    [ $? -eq 0 ] || exit $?
+    echo "python${PYTHON_VERSION}" iothub_client_map_test.py
+    "python${PYTHON_VERSION}" iothub_client_map_test.py
+    [ $? -eq 0 ] || exit $?
+    cd $build_root
+
+    cd $build_root/service/tests/
+    echo "python${PYTHON_VERSION}" iothub_service_client_ut.py
+    "python${PYTHON_VERSION}" iothub_service_client_ut.py
+    [ $? -eq 0 ] || exit $?
+    echo "python${PYTHON_VERSION}" iothub_service_client_map_test.py
+    "python${PYTHON_VERSION}" iothub_service_client_map_test.py
+    [ $? -eq 0 ] || exit $?
+    cd $build_root
+}
 
+create_packages()
```

Comment:
this needs error checking

Reply:
There a 2 operations in this function (copy and bdist_wheel) and both will output errors if any.
When we manually run the release script we always have to check the output for errors.